compiler/rustc_errors/src/json.rs

```diff
@@ -138,6 +138,24 @@ impl JsonEmitter {
     pub fn ui_testing(self, ui_testing: bool) -> Self {
         Self { ui_testing, ..self }
     }
+
+    fn emit(&mut self, val: EmitTyped<'_>) -> io::Result<()> {
+        if self.pretty {
+            writeln!(self.dst, "{}", serde_json::to_string_pretty(&val).unwrap())
+        } else {
+            writeln!(self.dst, "{}", serde_json::to_string(&val).unwrap())
```

Comment:
This can be serde_json's to_writer/to_writer_pretty

Reply:
Hm, that loses the `\n`.